spring-pulsar/src/main/java/org/springframework/pulsar/core/DefaultTopicResolver.java

```diff
@@ -71,17 +71,20 @@ public Map<Class<?>, String> getCustomTopicMappings() {
 
 	@Override
 	public Optional<String> resolveTopic(@Nullable String userSpecifiedTopic, Supplier<String> defaultTopicSupplier) {
-		return doResolveTopic(userSpecifiedTopic, null, defaultTopicSupplier);
+		if (StringUtils.hasText(userSpecifiedTopic)) {
```

Comment:
Nice catch here as this would otherwise have broken my recent change to the sender/producer to call through the TR as well.